### PR TITLE
Rich text editor branch merge into the main

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

PR to merge the RTE feature branch to main

# Why

# How Tested

Tests, bundle size test

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/SendBox.tsx [hunk 1]
```diff
@@ -243,18 +214,22 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
...
       return;
     }
 
     const message = textValue;
     // we don't want to send empty messages including spaces, newlines, tabs
     // Message can be empty if there is a valid file upload
-    if (!EMPTY_MESSAGE_REGEX.test(message) || hasFile(props)) {
+    if (
+      !EMPTY_MESSAGE_REGEX.test(message) ||
+      /* @conditional-compile-remove(file-sharing) */ hasCompletedFileUploads(activeFileUploads)
```
Comment (AmyL219) [inline packages/react-components/src/components/SendBox.tsx:231]: Have we made changes on this?

File: packages/react-components/src/components/SendBox.tsx [hunk 2]
```diff
@@ -243,18 +214,22 @@ export const SendBox = (props: SendBoxProps): JSX.Element => {
     }
 
     // Don't send message until all files have been uploaded successfully
+    /* @conditional-compile-remove(file-sharing) */
     setFileUploadsPendingError(undefined);
 
-    if (hasIncompleteFileUploads(props)) {
-      /* @conditional-compile-remove(file-sharing) */
+    /* @conditional-compile-remove(file-sharing) */
+    if (hasIncompleteFileUploads(activeFileUploads)) {
```
Comment (AmyL219) [inline packages/react-components/src/components/SendBox.tsx:221]: Did we made the change?

Comment (AmyL219) [reply packages/react-components/src/components/SendBox.tsx:221]: cc @palatter

Comment (vhuseinova-msft) [reply packages/react-components/src/components/SendBox.tsx:221]: Yes, it was part of the latest Patrick's PR

File: packages/react-components/src/components/SendBox.tsx (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/SendBox.tsx:231]: Yes, it was part of the latest Patrick's PR

File: packages/react-components/src/components/SendBox.tsx (hunk 2 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/SendBox.tsx:221]: https://github.com/Azure/communication-ui-library/pull/4034

File: packages/react-components/src/components/SendBox.tsx (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/SendBox.tsx:231]: https://github.com/Azure/communication-ui-library/pull/4034

File: packages/react-components/src/components/SendBox.tsx (hunk 2 above)
Comment (palatter) [reply packages/react-components/src/components/SendBox.tsx:221]: Yes was part of my refactor on sendbox.

File: packages/react-components/src/components/SendBox.tsx (hunk 1 above)
Comment (palatter) [reply packages/react-components/src/components/SendBox.tsx:231]: part of my refactor as well

**Review by palatter — APPROVED** [review]

**Review by AmyL219 — APPROVED** [review]

**Pull request merged**